packages/local_auth/local_auth_windows/windows/local_auth_plugin.cpp

```diff
@@ -0,0 +1,223 @@
+          Canceled;
+    }
+
+    auto consentResult =
+        co_await winrt::capture<Windows::Foundation::IAsyncOperation<
+            Windows::Security::Credentials::UI::UserConsentVerificationResult>>(
+            userConsentVerifierInterop,
+            &::IUserConsentVerifierInterop::RequestVerificationForWindowAsync,
+            hWnd, hReason);
+
+    WindowsDeleteString(hReason);
+
+    return consentResult;
+  }
+
+  Windows::Foundation::IAsyncOperation<
+      Windows::Security::Credentials::UI::UserConsentVerifierAvailability>
+  CheckAvailabilityAsync() override {
+    return Windows::Security::Credentials::UI::UserConsentVerifier::
+        CheckAvailabilityAsync();
+  }
+
+  // Disallow copy and move.
+  UserConsentVerifierImpl(const UserConsentVerifierImpl&) = delete;
+  UserConsentVerifierImpl& operator=(const UserConsentVerifierImpl&) = delete;
+
+ private:
+  // The provider for the root window to attach the dialog to.
+  std::function<HWND()> get_root_window_;
+};
+
+// static
+void LocalAuthPlugin::RegisterWithRegistrar(PluginRegistrarWindows* registrar) {
+  auto channel = std::make_unique<MethodChannel<EncodableValue>>(
+      registrar->messenger(), "plugins.flutter.io/local_auth_windows",
+      &StandardMethodCodec::GetInstance());
+
+  auto plugin = std::make_unique<LocalAuthPlugin>(
+      [registrar]() { return GetRootWindow(registrar->GetView()); });
+
+  channel->SetMethodCallHandler(
+      [plugin_pointer = plugin.get()](const auto& call, auto result) {
+        plugin_pointer->HandleMethodCall(call, std::move(result));
+      });
+
+  registrar->AddPlugin(std::move(plugin));
+}
+
+// Default constructor for LocalAuthPlugin
+LocalAuthPlugin::LocalAuthPlugin(std::function<HWND()> window_provider)
+    : user_consent_verifier_(std::make_unique<UserConsentVerifierImpl>(
+          std::move(window_provider))) {}
+
+LocalAuthPlugin::LocalAuthPlugin(
+    std::unique_ptr<UserConsentVerifier> user_consent_verifier)
+    : user_consent_verifier_(std::move(user_consent_verifier)) {}
+
+LocalAuthPlugin::~LocalAuthPlugin() {}
+
+void LocalAuthPlugin::HandleMethodCall(
+    const MethodCall<EncodableValue>& method_call,
+    std::unique_ptr<MethodResult<EncodableValue>> result) {
+  if (method_call.method_name().compare("authenticate") == 0) {
+    Authenticate(method_call, std::move(result));
+  } else if (method_call.method_name().compare("getAvailableBiometrics") == 0) {
+    GetAvailableBiometrics(std::move(result));
+  } else if (method_call.method_name().compare("isDeviceSupported") == 0) {
+    IsDeviceSupported(std::move(result));
+  } else {
+    result->NotImplemented();
+  }
+}
+
+// Starts authentication process
+winrt::fire_and_forget LocalAuthPlugin::Authenticate(
+    const MethodCall<EncodableValue>& method_call,
+    std::unique_ptr<MethodResult<EncodableValue>> result) {
+  auto reasonW = Utf16FromUtf8(GetArgument<std::string>(
```

Comment:
Names here also need to use C++ style guide naming (`reason`, `biometric_only`, etc.)

Reply:
Fixed.